clients/src/main/java/io/strimzi/http/consumer/HttpConsumerClient.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.logging.log4j.Logger;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class HttpConsumerClient implements ClientsInterface {
+
+    private static final Logger LOGGER = LogManager.getLogger(HttpConsumerClient.class);
+    private final HttpConsumerConfiguration configuration;
+    private int consumedMessages;
+    private HttpClient client;
+    private TracingHandle tracingHandle;
+    private HttpHandle httpHandle;
+    private final ScheduledExecutorService scheduledExecutor;
+
+    public HttpConsumerClient(Map<String, String> configuration) {
+        this.configuration = new HttpConsumerConfiguration(configuration);
+        this.consumedMessages = 0;
+        this.client = HttpClient.newHttpClient();
+        this.tracingHandle = TracingUtil.getTracing();
+        this.httpHandle = tracingHandle.createHttpHandle("receive-messages");
+        this.scheduledExecutor = Executors.newScheduledThreadPool(1, r -> new Thread(r, "http-consumer"));
+    }
+
+    @Override
+    public void run() {
+        LOGGER.info("Starting {} with configuration: \n{}", this.getClass().getName(), configuration.toString());
+
+        createConsumer();
+        subscribeToTopic();
+        scheduledExecutor.scheduleWithFixedDelay(this::checkAndReceiveMessages, 0, configuration.getPollInterval(), TimeUnit.MILLISECONDS);
+
+        awaitCompletion();
+    }
+
+    @Override
+    public void awaitCompletion() {
+        try {
+            long timeoutForOperations = configuration.getMessageCount() * configuration.getPollInterval() + Constants.DEFAULT_TASK_COMPLETION_TIMEOUT;
+            scheduledExecutor.awaitTermination(timeoutForOperations, TimeUnit.MILLISECONDS);
+
+            if (consumedMessages >= configuration.getMessageCount()) {
+                LOGGER.info("All messages successfully received");
+            } else {
+                LOGGER.error("Unable to correctly receive all messages");
+                throw new RuntimeException("Failed to receive all messages");
+            }
+        } catch (InterruptedException e) {
+            LOGGER.error("Failed to wait for task completion due to: {}", e.getMessage());
+            e.printStackTrace();
+        } finally {
+            if (!scheduledExecutor.isShutdown()) {
+                scheduledExecutor.shutdownNow();
+            }
+        }
+    }
+
+    private void checkAndReceiveMessages() {
+        if (consumedMessages >= configuration.getMessageCount()) {
+            scheduledExecutor.shutdown();
+        } else {
+            this.consumeMessages();
+        }
+    }
+
+    public void createConsumer() {
+        String message = "{\"name\":\"" + configuration.getClientId() + "\",\"format\":\"json\",\"auto.offset.reset\":\"earliest\"}";
```

Comment:
why is this hardcoded?

Reply:
It was hardcoded even before, we don't have any env variable for configuring it (I mean for the `auto.offset.reset`).
Until now it was used and we never needed to actually configure it.

I can add an env variable for that, otherwise I will create a constant.